platform/ios/src/MGLMapView.mm

```diff
@@ -4729,7 +4736,7 @@ - (void)notifyMapChange:(mbgl::MapChange)change
                             rect = annotationView.frame;
                         }
 
-                        if ( ! CGRectIntersectsRect(rect, self.frame)) {
+                        if ( ! CGRectIntersectsRect(rect, self.frame) && self.deselectWhenOutOfBounds) {
```

Comment:
I'm not convinced this property is needed. I would prefer to just remove these lines unless I can see a use case where deselect when out of bounds should be the default behavior.